drivers/w1/w1_ds2477_85_common.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2022 Thomas Stranger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Common functions for Maxim DS2477,DS2485 1-Wire Masters
+ */
+
+#include "w1_ds2477_85_common.h"
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/w1.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+
+LOG_MODULE_REGISTER(w1_ds2477_85, CONFIG_W1_LOG_LEVEL);
+
+int ds2477_85_write_port_config(const struct device *dev, uint8_t reg, uint16_t value)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	uint8_t buf[5] = {CMD_WRITE_W1_PORT_CONFIG, 3, reg};
```

Comment:
Can you give those magic numbers names? It's not directly obvious what they mean. Goes for multiple occasions.

Reply:
These magic numbers indicate the command length.
I'll abstract these into `CMD_xyz_LEN` and similar for all the I2C commands.
I believe I can pass the rx/tx-len of the script commands (`cfg->w1_script_cmd()`) directly, or would you abstract these as well?

Reply:
"ARRAY_SIZE" would be perfect but no super strong opinion on that